synapse/handlers/room_member.py

```diff
@@ -503,6 +503,50 @@ async def update_membership(
 
         return result
 
+    async def update_m_direct(
+        self,
+        sender: str,
+        target: str,
+        room_id: str,
+    ) -> None:
+        """Update a user's m_direct list.
```

Comment:
Note that this PR should ensure that there is a way of allowing the Module API to ignore updates from itself via the proposed `on_account_data_change` callback (https://github.com/matrix-org/synapse/pull/12327). Perhaps by adding an argument to that method to inform modules that it was a module vs. a user that updated the account data.